plugin-flex-ts-template-v2/src/feature-library/worker-details/custom-components/WorkerDetailsContainer/WorkerDetailsContainer.tsx

```diff
@@ -0,0 +1,173 @@
+import AttributeCustom from './AttributeCustom';
+
+interface OwnProps {
+  worker?: IWorker;
+}
+
+const WorkerDetailsContainer = ({ worker }: OwnProps) => {
+  const [changed, setChanged] = useState(false);
+  const [teamName, setTeamName] = useState('');
+  const [departmentName, setDepartmentName] = useState('');
+  const [otherAttributes, setOtherAttributes] = useState({} as any);
+
+  const textAttributes = getTextAttributes();
+  const booleanAttributes = getBooleanAttributes();
+
+  useEffect(() => {
+    if (worker) {
+      setTeamName(worker.attributes.team_name || '');
+      setDepartmentName(worker.attributes.department_name || '');
+      const other = {} as any;
+      Object.keys(worker.attributes).forEach((key: string) => {
+        if (textAttributes.includes(key)) {
+          other[key] = worker.attributes[key];
+        }
+        if (booleanAttributes.includes(key)) {
+          other[key] = worker.attributes[key] || false;
+        }
+      });
+      setOtherAttributes(other);
+    }
+  }, [worker]);
+
+  const handleTeamChange = (team: string) => {
+    setChanged(true);
+    setTeamName(team);
+  };
+
+  const handleDeptChange = (dept: string) => {
+    setChanged(true);
+    setDepartmentName(dept);
+  };
+
+  const handleOtherChange = (key: string, value: string) => {
+    setChanged(true);
+    const attributes = { ...otherAttributes, [key]: value };
+    setOtherAttributes(attributes);
+  };
+
+  const saveWorkerAttributes = async () => {
+    const workerSid = worker && worker.sid;
+    // Only save if worker was selected
+    if (workerSid) {
+      const updatedAttr = {
+        team_id: teamName,
+        team_name: teamName,
+        department_id: departmentName,
+        department_name: departmentName,
+        ...otherAttributes,
+      } as any;
+      await TaskRouterService.updateWorkerAttributes(workerSid, JSON.stringify(updatedAttr));
+      setChanged(false);
+    }
+  };
+
+  const SectionHeader = styled('div')`
+    flex: 0 0 auto;
+    font-size: 0.875rem;
+    font-weight: 700;
+    line-height: 1.25rem;
+    margin: 1.25rem 1rem 0.5rem;
+    padding: 0.5rem 0px;
+    border-bottom: 1px solid ${(props) => (props.theme as Theme).tokens.borderColors.colorBorderWeak};
+    color: ${(props) => (props.theme as Theme).tokens.textColors.colorText};
+  `;
+
+  return (
+    <Stack orientation="vertical" spacing="space0">
+      {isWorkerCanvasTabsEnabled() ? null : (
+        <SectionHeader>
+          <Template source={templates[StringTemplates.PSWorkerDetailsContainerName]} />
+        </SectionHeader>
+      )}
+      <Table variant="borderless">
+        <TBody>
+          <Tr key="agent_name">
+            <Td element="WORKER_DETAILS">
+              <Label htmlFor="name">
+                <Template source={templates.PSWorkerDetailsName} />
+              </Label>
+            </Td>
+            <Td element="WORKER_DETAILS">{worker?.fullName || 'Agent'}</Td>
+          </Tr>
```

Comment:
This name row seems redundant given the name is already at the top of the profile section.